src/uwtools/config.py

```diff
@@ -752,3 +753,17 @@ def create_config_obj(
                     raise ValueError(err_msg)
                 # Dump to file:
                 dump_method(path=outfile, cfg=config_obj)
+
+
+def export_variables(config_dict: dict, section_path: list, log: Logger) -> None:
```

Comment:
If `section_path` is being typed as `List[str]`, what is the reasoning for leaving `config` as `dict` and not `Dict[str, Any]`? Simply because it's a way to say "we know it may not _necessarily_ be turtles all the down"?

Reply:
Yes, I feel like it might give a false sense of assurance. I was just wondering if there's not a way to express what we really want somehow and found this: https://github.com/python/mypy/issues/731. I haven't had time to look into / test it yet, though. It might be the way to go, but possible out-of-scope for this PR unless you're feeling adventurous.